app/services/register_records_downloader.rb

```diff
@@ -19,6 +19,14 @@ def download_format(format)
     end
   end
 
+  def download_ods
+    headers = register.fields_array
+    data = register.records.where(entry_type: 'user').find_each.map do |r|
```

Comment:
you could use activerecord's `pluck` here to avoid instantiating active record objects for each record. The trade off is you can't then batch things using `find_each`, but in this case you're reading the whole dataset into memory anyway so I don't think it matters.